pkg/flowaggregator/clickhouseclient/clickhouseclient_test.go

```diff
@@ -364,3 +331,59 @@ func TestUpdateCH(t *testing.T) {
 		return (err == nil), nil
 	}), "timeout while waiting for second flow record to be committed (after DB connection update)")
 }
+
+func TestParseDatabaseURL(t *testing.T) {
+	testcases := []struct {
+		url           string
+		expectedProto protocol
+		expectedAddr  string
+		expectedErr   error
+	}{
+		{
+			url:           "tcp://127.0.0.1:9000",
+			expectedProto: protocolTCP,
+			expectedAddr:  "127.0.0.1:9000",
+			expectedErr:   nil,
+		},
+		{
+			url:           "abc://127.0.0.1:9000",
+			expectedProto: protocolUnknown,
+			expectedAddr:  "",
```

Comment:
you can omit this for error cases, it will be less verbose (applies to both `expectedProto` and `expectedAddr`)